src/fivetran-destination/src/destination/dml.rs

```diff
@@ -423,23 +435,68 @@ async fn delete_files(
         .context("update deletes")?;
     tracing::info!(?total_count, "altered rows in {qualified_table_name}");
 
+    // Clear our scratch table.
+    guard.clear().await.context("clearing guard")?;
+
     Ok(())
 }
 
-/// Creates a `TEMPORARY` table that will get dropped at the end of the SQL session, with the same
-/// format as the provided [`Table`].
-async fn create_temporary_table(
+#[must_use = "Need to clear the scratch table once you're done using it."]
+struct ScratchTableGuard<'a> {
+    client: &'a tokio_postgres::Client,
+    qualified_name: UnresolvedItemName,
+}
+
+impl<'a> ScratchTableGuard<'a> {
+    /// Deletes all the rows from the associated scratch table.
+    async fn clear(self) -> Result<(), OpError> {
+        let clear_table_stmt = format!("DELETE FROM {}", self.qualified_name);
+        let rows_cleared = self
+            .client
+            .execute(&clear_table_stmt, &[])
+            .await
+            .map_err(OpErrorKind::TemporaryResource)
+            .context("scratch table guard")?;
+        tracing::info!(?rows_cleared, table_name = %self.qualified_name, "guard cleared table");
+
+        Ok(())
+    }
+}
+
+/// Gets the existing "scratch" table that we can copy data into, or creates one if it doesn't
+/// exist.
+async fn get_scratch_table<'a>(
+    database: &str,
+    schema: &str,
     table: &Table,
```

Comment:
I'm a little hazy on the fivetran-destination. What is the `database`, `schema`, and `table` for?

Reply:
These are for the table we are syncing into, for example Fivetran will tell us to insert new rows into the `fivetran.hubspot.users` table

Reply:
And it's not possible to have to sinks pointing to the same table? I'm just trying to think through if collisions are possible and what would be the result.